cmd/influxd/upgrade/database.go

```diff
@@ -222,3 +193,45 @@ func upgradeDatabases(ctx context.Context, v1 *influxDBv1, v2 *influxDBv2, v1opt
 
 	return db2BucketIds, nil
 }
+
+// checkDiskSpace ensures there is enough room at the target path to store
+// a full copy of all V1 data.
+func checkDiskSpace(ui *input.UI, opts *options, log *zap.Logger) error {
+	log.Info("Checking available disk space")
+
+	size, err := DirSize(opts.source.dataDir)
+	if err != nil {
+		return fmt.Errorf("error getting size of %s: %w", opts.source.dataDir, err)
+	}
+
+	walSize, err := DirSize(opts.source.walDir)
+	if err != nil {
+		return fmt.Errorf("error getting size of %s: %w", opts.source.walDir, err)
+	}
+	size += walSize
+
+	v2dir := filepath.Dir(opts.target.boltPath)
+	diskInfo, err := fs.DiskUsage(v2dir)
+	if err != nil {
+		return fmt.Errorf("error getting info of disk %s: %w", v2dir, err)
+	}
+
+	freeBytes := humanize.Bytes(diskInfo.Free)
+	requiredBytes := humanize.Bytes(size)
+	log.Info("Computed disk space", zap.String("free", freeBytes), zap.String("required", requiredBytes))
+
+	if size > diskInfo.Free {
+		return fmt.Errorf("not enough space on target disk of %s: need %d, available %d", v2dir, size, diskInfo.Free)
+	}
+	if !opts.force {
+		if confirmed := internal.GetConfirm(ui, func() string {
+			return fmt.Sprintf(`Proceeding will copy all V1 data to %q
+  Space available: %s
+  Space required:  %s
+`, v2dir, freeBytes, requiredBytes)
```

Comment:
@psteinbachs @timhallinflux can you give a :+1: or :-1: to this behavior change? It adds one more y/N interaction point to `influxd upgrade`, to force users to consider how much disk they're about to use. The check is skipped if the `--force` flag is given.

Reply:
Yes, given community feedback around this I think it is a welcome change.